quickwit/quickwit-control-plane/src/control_plane.rs

```diff
@@ -103,23 +108,23 @@ impl ControlPlane {
         universe: &Universe,
         cluster_config: ClusterConfig,
         self_node_id: NodeId,
+        cluster_change_stream_factory: impl ClusterChangeStreamFactory,
```

Comment:
I don't understand why we need a factory here.


You seem to just use it to build the stream and then drop the factory.
Can we use a `ClusterChangeStream` directly here?

Secondary question, can we drop the concept of `ClusterChangeStreamFactory` entirely?
Same for `ClusterChangeStreamFactoryForTest`

Reply:
Answering my own question:

It is not directly built. We pass it into the supervise_fn closure which can be called several times.